clouds/snowflake/modules/sql/h3/H3_TOPARENT.sql

```diff
@@ -1,30 +1,15 @@
 ----------------------------
--- Copyright (C) 2021 CARTO
+-- Copyright (C) 2021-2024 CARTO
 ----------------------------
 
-CREATE OR REPLACE FUNCTION @@SF_SCHEMA@@._H3_TOPARENT
-(index STRING, resolution DOUBLE)
-RETURNS STRING
-LANGUAGE JAVASCRIPT
-IMMUTABLE
-AS $$
-    if (!INDEX) {
-        return null;
-    }
-
-    @@SF_LIBRARY_H3_TOPARENT@@
-
-    if (!h3ToparentLib.h3IsValid(INDEX)) {
-        return null;
-    }
-
-    return h3ToparentLib.h3ToParent(INDEX, Number(RESOLUTION));
-$$;
-
 CREATE OR REPLACE SECURE FUNCTION @@SF_SCHEMA@@.H3_TOPARENT
-(index STRING, resolution INT)
+(h3_hex STRING, resolution INT)
 RETURNS STRING
 IMMUTABLE
 AS $$
-    @@SF_SCHEMA@@._H3_TOPARENT(INDEX, CAST(RESOLUTION AS DOUBLE))
+    IFF(
```

Comment:
The same way that we have added a condition to `H3_TOCHILDREN` to see if the parameter resolution is bigger. The same could be applied here but with smaller resolutions.

It might be also ok in both cases to check that the resolution is in [0,15], as done in `H3_FROMLONGLAT`.